File: server/src/main/java/com/krillsson/sysapi/resources/EventResource.java
package com.krillsson.sysapi.resources;

import com.krillsson.sysapi.auth.BasicAuthorizer;
import com.krillsson.sysapi.config.UserConfiguration;
import com.krillsson.sysapi.core.monitoring.MonitorManager;
import com.krillsson.sysapi.core.monitoring.MonitorMapper;
import com.krillsson.sysapi.dto.monitor.MonitorEvent;
import io.dropwizard.auth.Auth;

import javax.annotation.security.RolesAllowed;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

import static javax.ws.rs.core.Response.Status.NOT_FOUND;

@Path("events")
@Produces(MediaType.APPLICATION_JSON)
public class EventResource {
    private final MonitorManager monitorManager;

    public EventResource(MonitorManager monitorManager) {
        this.monitorManager = monitorManager;
    }

    @GET
    @RolesAllowed(BasicAuthorizer.AUTHENTICATED_ROLE)
    public List<MonitorEvent> getRoot(@Auth UserConfiguration user) {
        return MonitorMapper.INSTANCE.map(monitorManager.events());
    }

    @DELETE
    @Path("{id}")
    @RolesAllowed(BasicAuthorizer.AUTHENTICATED_ROLE)
    public void delete(@Auth UserConfiguration user, @PathParam("id") String id) {
        boolean removed = monitorManager.removeEvents(id);
        if (!removed) {
            throw new WebApplicationException(NOT_FOUND);
        }
    }
}

File: server/src/main/java/com/krillsson/sysapi/rest/EventResource.kt
package com.krillsson.sysapi.rest

import com.krillsson.sysapi.auth.BasicAuthorizer
import com.krillsson.sysapi.core.domain.event.Event
import com.krillsson.sysapi.core.monitoring.event.EventManager
import java.util.*
import javax.annotation.security.RolesAllowed
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("events")
@Produces(MediaType.APPLICATION_JSON)
@RolesAllowed(BasicAuthorizer.AUTHENTICATED_ROLE)
class EventResource(private val eventManager: EventManager) {
    @GET
    @RolesAllowed(BasicAuthorizer.AUTHENTICATED_ROLE)
    fun getRoot(): List<Event> {
        return eventManager.getAll()
    }

    @DELETE
    @Path("{id}")
    @RolesAllowed(BasicAuthorizer.AUTHENTICATED_ROLE)
    fun delete(@PathParam("id") id: String?) {
        val removed = eventManager.remove(UUID.fromString(id))
        if (!removed) {
            throw WebApplicationException(Response.Status.NOT_FOUND)
        }
    }
}